mvpa2/tests/test_generators.py

```diff
@@ -221,6 +241,24 @@ def test_balancer():
     # now use it as a generator
     dses = list(bal.generate(ds))
     assert_equal(len(dses), 5)
+
+    # if we rerun again, it would be a different selection
+    res2 = bal(ds)
+    assert_true(np.any(res.sa.ids != bal(ds).sa.ids))
+
+    # but if we create a balancer providing seed rng int,
+    # should be identical results
+    bal = Balancer(apply_selection=True, count=5, rng=1)
+    assert_false(np.any(bal(ds).sa.ids != bal(ds).sa.ids))
+
+    # But results should differ if we use .generate to produce those multiple
+    # balanced datasets
+    b = Balancer(apply_selection=True, count=3, rng=1)
+    balanced = list(b.generate(ds))
+    assert_false(all(balanced[0].sa.ids == balanced[1].sa.ids))
+    assert_false(all(balanced[0].sa.ids == balanced[2].sa.ids))
+    assert_false(all(balanced[1].sa.ids == balanced[2].sa.ids))
```

Comment:
In addition to the sequence not being constant, I would check that the sequence is repeated on a new `generate` call:

```Python
assert_true(all(np.array_equal(ds_a.sa.ids, ds_b.sa.ids)
            for ds_a, ds_b in zip(balanced, b.generate(ds)))
```